aeneas/src/jvm/JvmHeap.v3

```diff
@@ -100,7 +100,10 @@ class JvmHeap(jprog: JvmProgram) {
 			TUPLE => jprog.context.fail("normalization should remove all tuple values");
 			VARIANT,
 			CLASS,
-			ARRAY => visitRecord(Record.!(val), etype, ref, index);
+			ARRAY => match (val) {
```

Comment:
As far as I can tell, I think this wouldn't have been hit before this change, because there was never an instance of JVM code pointer becoming cast to `AnyRef.TYPE`. Then, it would've fallen through to the `instrs.put(JvmHI_Value.new(val, etype))` case directly.

Now that we have this potential cast, it incorrectly tries to visit it as a record because the `kind` of `AnyRef.TYPE` is `CLASS`. I had to add this here to fix that so it wouldn't try to cast it as a record, instead relying on `val`'s type. Do you have a better workaround?